WebChannelAPI.md

```diff
@@ -0,0 +1,474 @@
+    ```
+
+    * Supported Events:
+
+    |event|action
+    |---|---
+    |`co.meldstudio.events.screenshot`|Takes a screenshot.
+
+12. ## **`setStagedScene(sceneId: string)`**
+
+    Prepares a scene for switching without making it live.
+
+    **Parameters**:
+    |property|type|description
+    |---|---|---
+    |sceneId|`string`|Scene Identifier of the scene to mark as staged.
+
+    ```js
+    const sceneId = /* desired scene id */;
+    meld.setStagedScene(sceneId);
+    ```
+
+13. ## **`showStagedScene()`**
+
+    Switches to the prepared staged scene.
+
+## Session Schema
+
+The `session` property contains a JSON object with a single object with a key of `items`. Item's contains all scenes, layers, and audio tracks in the current Meld Studio session, and their relevant properties.
+
+Each item is stored in the "items" object by it's identifier: `/^[ 0-9A-F ]{ 32 }$/`. These are guaranteed to be unique within a given session.
+```json
+{
+    "items": {
+        "1DA03C46454676F85DD8CCB1EF3CBF83": {
+            "current": true,
+            "index": 0,
+            "name": "Playing",
+            "staged": false,
+            "type": "scene"
+        },
+        "3B8EFA69F29E81C290B5BD09CC2EA48E": {
+            "enabled": true,
+            "name": "Corner Pin",
+            "parent": "DD8FD6181E7560B45D75B79F005B0299",
+            "type": "effect"
+        },
+        "DD8FD6181E7560B45D75B79F005B0299": {
+            "index": 0,
+            "name": "Fortnite",
+            "parent": "1DA03C46454676F85DD8CCB1EF3CBF83",
+            "type": "layer",
+            "visible": true
+        },
+        "F1BA59F85C196B9088A06445BE6E5E20": {
+            "monitoring": false,
+            "muted": false,
+            "name": "Fortnite",
+            "parent": "DD8FD6181E7560B45D75B79F005B0299",
+            "type": "track"
+        },
+        "F32F720ACA7B902079E748A33D0113F8": {
+            "monitoring": false,
+            "muted": false,
+            "name": "Track 1",
+            "type": "track"
+        }
+    }
+}
+```
+
+
+### Session Properties
+
+#### Scene:
+
+|property|type|description
+|---|---|---
+|type|`string`|`scene`
+|index|`int`|Position of the Scene in the Scene list.
+|name|`string`|User specified name of the scene.
+|current|`bool`|Whether the scene is currently being displayed.
+|staged|`bool`|Whether the scene is staged to be presented.
+
+```json
+{
+    "current": true,
+    "index": 0,
+    "name": "Intro",
+    "staged": false,
+    "type": "scene"
+}
+```
+
+#### Audio Track:
+
+|property|type|description
+|---|---|---
+|type|`string`|`track`
+|parent|`string`|`<optional>` Scene Identifier; if not present, the track is a global track.
+|name|`string`|If `parent` is set, this mirrors the name of the parent `Layer`. If not set, this has the name of the track.
+|monitoring|`bool`|Whether the track is currently cued.
+|muted|`bool`|Whether the track is currently muted.
+
+
+##### Global Track:
```

Comment:
the formatting for these two blocks can be fixed by removing the indent.

Reply:
Good catch. I didn't even notice.

Reply:
Fixed in 7d676dc136ad0ef8390ce78eeca8b749bd479623